### DISCUSSION TITLE
Unit tests failing on master?

### DISCUSSION THREAD

**Discussion by dmathijs** [st3v3nmw/obsidian-spaced-repetition#757] [General]
Wanted to have a crack at going through the code so I can help a bit..
I ran `npm i` followed by `npm test` and I'm getting a lot of failing tests, am I doing something wrong?

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/4018247/cb110c1f-8d13-48ad-ade7-bb35d786619f)

**Comment by ronzulu** [comment]
Hi  @dmathijs 

At one stage this project was using npm, but did switch to pnpm.

I don't know if that could make a difference, but the following worked for me just now:

```
pnpm install
pnpm test
```

![image](https://github.com/st3v3nmw/obsidian-spaced-repetition/assets/75528127/084026b3-e54e-451d-9911-56415813c41f)

Cheers
Ronny

**Reply by dmathijs** [reply]
Hi Ronny,

https://github.com/st3v3nmw/obsidian-spaced-repetition/blob/f20e9e34507c25c0b5f079ec4b40582491075dba/package.json#L54C32-L54C32 clearly indicates that package manager is supposed to be 'npm', the docs indeed reference pnpm but it seems like this was reverted 10 months ago in: https://github.com/st3v3nmw/obsidian-spaced-repetition/commit/42c3796709ea33d1740e5651ac4ddb23a97be297. I tried running with pnpm but it raises some exception saying that the project was configured for use with npm. So it looks to me like there's something off here.

**Comment by ronzulu** [comment]
Hi Diederik, I'll just loop in the package owner.

Hi @st3v3nmw, you might be best placed to answer about the package manager

Diederik, I cloned https://github.com/st3v3nmw/obsidian-spaced-repetition into a fresh folder locally. Then I ran `npm i` followed by `npm test` (not doing anything else) and all test ran correctly.

Are you able to attach the detailed error output from the tests?

FYI I'm using vscode.

Cheers
Ronny

**Comment by dmathijs** [comment]
Hi Ronny,

After some investigating I noticed that I was using Node version 14.18.2, some of the string methods used in tests are not supported by that version which caused tests to fail, I upgraded to 18.18.1 and it worked! Thanks for you help!

**Comment by ronzulu** [comment]
Great that you solved it, thanks for letting me know!
